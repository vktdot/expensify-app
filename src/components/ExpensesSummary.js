import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import selectExpensesTotal from '../selectors/expenses-total'
import  getVisibleExpenses from '../selectors/expenses'

export const ExpensesSummary = (props) => {
    const expenseWord = props.expenseCount === 1? 'expense': 'expenses';
    const formattedTotal = numeral(props.expensesTotal/100).format('$0,0.00')
    return(<h3>Viewing {props.expenseCount} {expenseWord} totalling {formattedTotal}
    </h3>)
    
}

const MapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}
    
    
export default connect(MapStateToProps)(ExpensesSummary)