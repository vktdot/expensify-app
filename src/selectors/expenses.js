import moment from 'moment';

export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const createdAtMoment = moment(expense.createdAt)
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true
        const endDateMatch = endDate? endDate.isSameOrAfter(createdAtMoment, 'day') : true
        const textMatch = typeof text!== 'string' || expense.description.toLowerCase().includes(text.toLowerCase())  ;
        return startDateMatch && endDateMatch && textMatch;
    }).sort(function(a,b){
        if(sortBy === 'date'){
            return b.createdAt - a.createdAt
        }
        else{
            return b.amount - a.amount
        }
    })
}