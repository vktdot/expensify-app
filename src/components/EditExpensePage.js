import React from 'react';
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { editExpense, startRemoveExpense, startEditExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit= (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }
    onRemove = () =>{
        this.props.startRemoveExpense(this.props.expense.id);
        this.props.history.push('/')
    } 
    render(){
        return(
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm 
                    onSubmit = {this.onSubmit}
                    expense = {this.props.expense}
                    />
                    <button className = "button button--secondary" onClick = {this.onRemove}>
                        Remove Expense
                     </button>
                </div>
            </div>
        )
    }
}

// const EditExpensePage = (props) => {
//         return(
//             <div>
//                 <ExpenseForm 
//                   expense = {props.expense}
//                   onSubmit={(expense) => {
//                       props.dispatch(editExpense(expense.id, expense));
//                       props.history.push('/');
//                   }}
//                 />
//                 <button onClick = {() =>{
//                     props.dispatch(removeExpense(props.expense.id));
//                     props.history.push('/')}}  >
//                   Remove
//                 </button>
//             </div>
//         )
// }

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
})
    
const mapDispatchToProps = (dispatch) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (expense) => dispatch(startRemoveExpense(expense))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);