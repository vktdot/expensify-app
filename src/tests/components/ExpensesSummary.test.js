import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'

test('should generate summary for 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses = {expenses[1]}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Should generate summary for 0 or multiple expenses', ()=>{
    const wrapper = shallow(<ExpensesSummary expenses = {expenses}/>)
    expect(wrapper).toMatchSnapshot()
})