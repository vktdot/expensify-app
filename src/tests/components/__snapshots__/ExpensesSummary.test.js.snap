// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should generate summary for 0 or multiple expenses 1`] = `
<h3>
  Viewing 
   
  expenses
   totalling 
  $0.00
</h3>
`;

exports[`should generate summary for 1 expense 1`] = `
<h3>
  Viewing 
   
  expenses
   totalling 
  $0.00
</h3>
`;
